node/npm_dependencies.ts

```diff
@@ -109,27 +174,34 @@ const getNPMSpecifiers = async (basePath: string, functions: string[], importMap
         }
       })
     }
-  })
+  }
 
   return {
-    npmSpecifiers: [...npmSpecifiers],
+    npmSpecifiers,
     npmSpecifiersWithExtraneousFiles: [...npmSpecifiersWithExtraneousFiles],
   }
 }
 
+const prependFile = async (path: string, prefix: string) => {
+  const existingContent = await fs.readFile(path, 'utf8')
```

Comment:
I really don't like that we are writing a file to disk, then loading it again into memory, and then writing it back to disk.

I think in the future it might be nice to set `write: false` in esbuild, so that we get the file contents directly and write to disk just once.

Reply:
Could we create an issue so we do this as a follow-up?

Reply:
Done in https://github.com/netlify/edge-bundler/pull/505/commits/a8880e05d0913d44139fa5fe9e4f9e2b467396c7